hack/scale_out_poc/config_haproxy/setup_haproxy.sh

```diff
@@ -0,0 +1,158 @@
+        echo 
+        echo "To run it: "
+        echo "    TENANT_PARTITION_IP=\"##.##.##.##,##.##.##.##...\" RESOURCE_PARTITION_IP=##.##.##.## setup_haproxy.sh"
+}
+
+run_command_exit_if_failed() {
+        command="$@"
+        $command 
+        if [[ $? != 0 ]] 
+        then 
+                printf "\033[0;31mFailed: ${command}\n"
+                printf "Exit. \033[0m\n"
+                exit 1
+        fi
+}
+
+install_haproxy_if_needed() {
+        haproxy -v > /dev/null 2>&1
+        if [[ $? != 0 ]] 
+        then 
+                printf "\033[0;31mHaproxy does not exist, installing haproxy...\033[0m\n"
+                run_command_exit_if_failed sudo apt-get update && sudo apt-get -y upgrade
+                run_command_exit_if_failed sudo apt --assume-yes install haproxy
+        fi
+}
+
+patch-haproxy-prometheus() {
+  # based on https://www.haproxy.com/blog/haproxy-exposes-a-prometheus-metrics-endpoint
+  echo 'Patching Haproxy to expose prometheus...'
+  haproxy_patch_tmp=/tmp/haproxy
+  run_command_exit_if_failed sudo apt install -y git ca-certificates gcc libc6-dev liblua5.3-dev libpcre3-dev libssl-dev libsystemd-dev make wget zlib1g-dev
+  run_command_exit_if_failed git clone https://github.com/haproxy/haproxy.git ${haproxy_patch_tmp}
+  run_command_exit_if_failed pushd ${haproxy_patch_tmp}
+  run_command_exit_if_failed git checkout tags/v2.3.0
+  run_command_exit_if_failed make TARGET=linux-glibc USE_LUA=1 USE_OPENSSL=1 USE_PCRE=1 USE_ZLIB=1 USE_SYSTEMD=1 EXTRA_OBJS=contrib/prometheus-exporter/service-prometheus.o -j4
+  run_command_exit_if_failed sudo make install-bin
+  run_command_exit_if_failed sudo systemctl reset-failed haproxy.service
+  run_command_exit_if_failed sudo systemctl stop haproxy
+  run_command_exit_if_failed sudo cp /usr/local/sbin/haproxy /usr/sbin/haproxy
+  run_command_exit_if_failed sudo systemctl start haproxy
+  run_command_exit_if_failed haproxy -vv|grep Prometheus
+  run_command_exit_if_failed popd
+  run_command_exit_if_failed rm -rf ${haproxy_patch_tmp}
+}
+
+function direct-haproxy-logging {
+  haproxy_conf=`find /etc/rsyslog.d -name *haproxy.conf`
+
+  if [ -z "$haproxy_conf" ]; then
+    echo "haproxy conf file not found in /etc/rsyslog.d/"
+    return
+  fi
+
+  if grep -q "UDPServerRun 514" "$haproxy_conf"; then
+    echo "skipped updating haproxy.conf for directing logging"
+    return
+  fi
+
+   echo '
+$ModLoad imudp
+$UDPServerRun 514
+local0.* -/var/log/haproxy.log
+' >> $haproxy_conf
+
+  run_command_exit_if_failed sudo service rsyslog restart
+  echo "haproxy logging directed to /var/log/haproxy.log only"
+}
+
+config_haproxy() {
+        script_root=$(dirname "${BASH_SOURCE}")
+        local -r temp_haproxy_cfg="/tmp/haproxy.cfg"
+
+        export GO111MODULE=on
+        script_root=$(dirname "${BASH_SOURCE}")
+        repo_root=$(cd $(dirname $0)/../../.. ; pwd)
+
+        template_source=${repo_root}/hack/scale_out_poc/config_haproxy/haproxy.cfg.template
```

Comment:
An updated version of haproxy.cfg.template has already been moved to a different location in master branch in PR #929.  Could you update it accordingly?

Reply:
deleted